File: code/functions/function-type.js
// Function types in JavaScript
function add(a, b) {
    return a + b;
}

function multiply(a, b) {
    return a * b;
}

function applyOperation(op, x, y) {
    return op(x, y);
}

let operation = add;
let result = applyOperation(operation, 5, 3);
console.log("Result: " + result);

operation = multiply;
result = applyOperation(operation, 5, 3);
console.log("Result: " + result);

// Or with arrow functions
const arrowAdd = (a, b) => a + b;
result = applyOperation(arrowAdd, 5, 3);
console.log("Result: " + result);

File: code/functions/function-type.swift
func add(_ a: Int, _ b: Int) -&gt; Int {
    return a + b
}

func multiply(_ a: Int, _ b: Int) -&gt; Int {
    return a * b
}

func applyOperation(_ op: (Int, Int) -&gt; Int, _ x: Int, _ y: Int) -&gt; Int {
    return op(x, y)
}

var operation: (Int, Int) -&gt; Int = add
var result = applyOperation(operation, 5, 3)
print("Result: \(result)")

operation = multiply
result = applyOperation(operation, 5, 3)
print("Result: \(result)")